noxfile.py

```diff
@@ -110,9 +110,20 @@ def __call__(self, frag):
             )
 
             # Install the generated fragment library.
-            # Note: install into the tempdir to prevent issues
-            # with running pip concurrently.
-            self.session.install(tmp_dir, "-e", ".", "-t", tmp_dir, "-qqq")
+            if self.use_ads_templates:
+                self.session.install(tmp_dir, "-e", ".", "-qqq")
+            else:
+                # Use the constraints file for the specific python runtime version
```

Comment:
Why don't we need all the protobuf run-time logic in the ads-templates as well? Could you add a note in the PR description?

Reply:
Fixed in [e689b13](https://github.com/googleapis/gapic-generator-python/pull/1749/commits/e689b136d48371a6cd736c10c44341b31b6aee96)